src/Evolution/Systems/GrMhd/ValenciaDivClean/FixConservatives.cpp

```diff
@@ -205,25 +227,42 @@ bool FixConservatives::operator()(
                        ::Tags::TempScalar<3>>>
       temp_buffer(size);
 
-  Scalar<DataVector>& tilde_b_squared = get<::Tags::TempScalar<1>>(temp_buffer);
-  dot_product(make_not_null(&tilde_b_squared), tilde_b, tilde_b,
-              spatial_metric);
-
   Scalar<DataVector>& tilde_s_squared = get<::Tags::TempScalar<2>>(temp_buffer);
   dot_product(make_not_null(&tilde_s_squared), *tilde_s, *tilde_s,
               inv_spatial_metric);
 
+  const bool non_zero_mag =
+      magnetic_field_treatment_ == hydro::MagneticFieldTreatment::AssumeNonZero
+          ? true
+          : (magnetic_field_treatment_ !=
+             hydro::MagneticFieldTreatment::AssumeZero);
```

Comment:
Equivalent: `const bool non_zero_mag = magnetic_field_treatment_ != hydro::MagneticFieldTreatment::AssumeZero` (or `== AssumeNonZero` since you overwrite it in the third case).